player.rb

```diff
@@ -1,16 +1,56 @@
 require 'gosu'
+require 'chipmunk'
+require_relative 'chip-gosu-functions'
+require_relative 'obj'
 
-class Player
-  def initialize(window)
-    @image = Gosu::Image.new window, "resources/images/ball.png"
-    @x, @y = 100, 100
+SUBSTEPS = 6
+SCREEN_WIDTH = 640
+SCREEN_HEIGHT = 480
+class Player < Obj
+  attr_accessor :j
+  def initialize(window, g, diameter)
+    super window, g, "resources/images/ball.png"
+
+    @diameter = diameter
+
+    @body = CP::Body.new(20.0, 150.0)
+    @shapes << CP::Shape::Circle.new(body, diameter / 2 , CP::Vec2.new(0,0))
+    @shapes.each do |shape|
+      shape.body.p = CP::Vec2.new(0.0, 0.0)
+      shape.body.v = CP::Vec2.new(0.0, 0.0)
+      shape.e = 1
+      shape.body.a = (3*Math::PI/2.0)
```

Comment:
Wow :D you really hate parantheses and my bad spacing. :D I'll do something soon.